controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -298,8 +298,9 @@ func (r *MultiClusterObservabilityReconciler) initFinalization(
 		log.Info("Finalizer removed from mco resource")
 		return true, nil
 	}
-	if !util.Contains(mco.GetFinalizers(), certFinalizer) {
-		mco.SetFinalizers(append(mco.GetFinalizers(), certFinalizer))
+	if !util.Contains(mco.GetFinalizers(), resFinalizer) {
+		mco.SetFinalizers(util.Remove(mco.GetFinalizers(), certFinalizer))
```

Comment:
Do we need to clean the old certificate, issuer... in created in old versions before removing the `certFinalizer`?

Reply:
@morvencao 
yes, need to consider it. maybe not delete, but reuse.
we will handle it in upgrade issue.